src/test/java/org/kohsuke/github/GHEventPayloadTest.java

```diff
@@ -0,0 +1,248 @@
+//        assertThat(page.getSha(), is("91ea1bd42aa2ba166b86e8aefe049e9837214e67"));
+//        assertThat(page.getHtmlUrl(), is("https://github.com/baxterthehacker/public-repo/wiki/Home"));
+//        assertThat(event.getRepository().getName(), is("public-repo"));
+//        assertThat(event.getRepository().getOwner().getLogin(), is("baxterthehacker"));
+//        assertThat(event.getSender().getLogin(), is("baxterthehacker"));
+//    }
+
+    @Test
+    public void issue_comment() throws Exception {
+        GHEventPayload.IssueComment event =
+                GitHub.offline().parseEventPayload(payload.asReader(), GHEventPayload.IssueComment.class);
+        assertThat(event.getAction(), is("created"));
+        assertThat(event.getIssue().getNumber(), is(2));
+        assertThat(event.getIssue().getTitle(), is("Spelling error in the README file"));
+        assertThat(event.getIssue().getState(), is(GHIssueState.OPEN));
+        assertThat(event.getIssue().getLabels().size(), is(1));
+        assertThat(event.getIssue().getLabels().iterator().next().getName(), is("bug"));
+        assertThat(event.getComment().getUser().getLogin(), is("baxterthehacker"));
+        assertThat(event.getComment().getBody(), is("You are totally right! I'll get this fixed right away."));
+        assertThat(event.getRepository().getName(), is("public-repo"));
+        assertThat(event.getRepository().getOwner().getLogin(), is("baxterthehacker"));
+        assertThat(event.getSender().getLogin(), is("baxterthehacker"));
+    }
+
+// TODO implement support classes and write test
+//    @Test
+//    public void issues() throws Exception {}
+
+// TODO implement support classes and write test
+//    @Test
+//    public void label() throws Exception {}
+
+// TODO implement support classes and write test
+//    @Test
+//    public void member() throws Exception {}
+
+// TODO implement support classes and write test
+//    @Test
+//    public void membership() throws Exception {}
+
+// TODO implement support classes and write test
+//    @Test
+//    public void milestone() throws Exception {}
+
+// TODO implement support classes and write test
+//    @Test
+//    public void page_build() throws Exception {}
+
+    @Test
+    @Payload("public")
+    public void public_() throws Exception {
+        GHEventPayload.Public event =
+                GitHub.offline().parseEventPayload(payload.asReader(), GHEventPayload.Public.class);
+        assertThat(event.getRepository().getName(), is("public-repo"));
+        assertThat(event.getRepository().getOwner().getLogin(), is("baxterthehacker"));
+        assertThat(event.getSender().getLogin(), is("baxterthehacker"));
+    }
+
+    @Test
+    public void pull_request() throws Exception {
+        GHEventPayload.PullRequest event =
+                GitHub.offline().parseEventPayload(payload.asReader(), GHEventPayload.PullRequest.class);
+        assertThat(event.getAction(), is("opened"));
```

Comment:
That runs the risk of following requests for the partial objects that are not available from the event objects. safer to just test the paths